service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -173,6 +201,16 @@ void testGetMetadata() {
     verify(datasetDao).retrieve(datasetId);
   }
 
+  @Test
+  void testGetMetadataWithPhsId() {
+    when(datasetDao.retrieve(tdrDataset.id())).thenReturn(tdrDataset);
+    when(samService.hasGlobalAction(user, SamAction.READ_ANY_METADATA)).thenReturn(true);
+    when(datarepoService.getSnapshotPhsId(user, tdrDataset.storageSourceId())).thenReturn("1234");
+    String datasetMetadata = datasetService.getMetadata(user, tdrDataset.id());
+    verify(datasetDao).retrieve(tdrDataset.id());
```

Comment:
Is this `verify()` needed? The test should fail if no mocks are called, and since this test has an `assert`, sonar won't complain about the test not having any asserts.